flocker/node/test/test_deploy.py

```diff
@@ -1159,19 +1161,47 @@ def test_proxy_needs_creating(self):
 
         nodes = frozenset([
             Node(
-                hostname=expected_destination_host,
+                uuid=destination_node_uuid,
                 applications=frozenset([application])
             )
         ])
 
         desired = Deployment(nodes=nodes)
+        current = DeploymentState(nodes=[
+            NodeState(uuid=destination_node_uuid,
+                      hostname=expected_destination_host)])
         result = api.calculate_changes(
-            desired_configuration=desired, current_cluster_state=EMPTY_STATE)
+            desired_configuration=desired, current_cluster_state=current)
         proxy = Proxy(ip=expected_destination_host,
                       port=expected_destination_port)
         expected = sequentially(changes=[SetProxies(ports=frozenset([proxy]))])
         self.assertEqual(expected, result)
 
+    def test_no_proxy_if_node_state_unknown(self):
+        """
+        ``ApplicationNodeDeployer.calculate_changes`` does not attempt to
+        create a proxy to a node whose state is unknown, since the
+        destination IP is unavailable.
+        """
+        api = ApplicationNodeDeployer(u'192.168.1.1', node_uuid=uuid4(),
```

Comment:
Eh, it's internal tests, not going to bother (but will try to remember in the future). Re routing... it's a problem, yes. Real solution is using a better routing system.